atomicapp/nulecule/main.py

```diff
@@ -183,6 +184,11 @@ def run(self, cli_provider, answers_output, ask,
         # it does exist it will be just be loaded and returned
         self.nulecule = self.unpack(dryrun=dryrun, config=self.answers)
 
+        # If no provider is specified within answers.conf or --provider=
+        # we should try and get a default one from the artifacts folder
+        if self.answers[GLOBAL_CONF][PROVIDER_KEY] is None and cli_provider is None:
+            self.answers[GLOBAL_CONF][PROVIDER_KEY] = Utils.getDefaultArtifact(self.app_path)
+
         self.nulecule.load_config(config=self.nulecule.config, ask=ask)
         self.nulecule.render(cli_provider, dryrun)
```

Comment:
me too, we should be determining cli_provider much earlier inthe code rather than at multiple points in `lib.py/main.py/base.py`.